src/test/java/org/truffleruby/PolyglotIOTest.java

```diff
@@ -9,64 +9,48 @@
  */
 package org.truffleruby;
 
-import com.oracle.truffle.api.source.Source;
-import com.oracle.truffle.api.vm.PolyglotEngine;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Source;
 import org.junit.Test;
+import org.truffleruby.launcher.Launcher;
+
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
+import java.io.IOException;
 
 import static org.junit.Assert.assertEquals;
 
 public class PolyglotIOTest extends RubyTest {
 
     @Test
-    public void testPolyglotIn() {
+    public void testPolyglotIn() throws IOException {
         final ByteArrayOutputStream out = new ByteArrayOutputStream();
         final ByteArrayInputStream in = new ByteArrayInputStream("abc".getBytes());
 
-        final PolyglotEngine engine = RubyTest.setupConfig(PolyglotEngine.newBuilder())
-                .setOut(out)
-                .setIn(in)
-                .build();
-
-        try {
-            engine.eval(Source.newBuilder("puts STDIN.read(3)").name("test.rb").mimeType(RubyLanguage.MIME_TYPE).build());
-        } finally {
-            engine.dispose();
+        try (Context context = Context.newBuilder().out(out).in(in).build()) {
+            context.eval(Source.newBuilder(Launcher.LANGUAGE_ID, "puts STDIN.read(3)", "test.rb").build());
```

Comment:
There is [`Source.create(lang, code)`](http://www.graalvm.org/truffle/javadoc/org/graalvm/polyglot/Source.html#create-java.lang.String-java.lang.CharSequence-) and [`Source.Builder.buildLiteral()`](http://www.graalvm.org/truffle/javadoc/org/graalvm/polyglot/Source.Builder.html#buildLiteral--) which do not throw IOException and are made for building literal Sources.